supervisely/api/image_api.py

```diff
@@ -3223,10 +3228,145 @@ def upload_multiview_images(
+            with sly.ApiContext(api, project_id=project_id, project_meta=project_meta):
+                for group_name, ids in enumerate(sly.batched(image_ids, batch_size=BATCH_SIZE)):
+                    api.image.group_images_for_multiview(ids, group_name, multiview_tag_name)
+
+        """
+
+        # ============= Api Context ===================================
+        # using context for optimization (avoiding extra API requests)
+        context = self._api.optimization_context
+        project_meta = context.get("project_meta")
+        project_id = context.get("project_id")
+        if project_id is None:
+            project_id = self.get_project_id(image_ids[0])
+            context["project_id"] = project_id
+        if project_meta is None:
+            project_meta = ProjectMeta.from_json(
+                self._api.project.get_meta(project_id, with_settings=True)
+            )
+            context["project_meta"] = project_meta
+        # =============================================================
+
+        need_update_project_meta = False
+        multiview_tag_name = multiview_tag_name or _MULTIVIEW_TAG_NAME
+        multiview_tag_meta = project_meta.get_tag_meta(multiview_tag_name)
+
+        if multiview_tag_meta is None:
+            multiview_tag_meta = TagMeta(
+                multiview_tag_name,
+                TagValueType.ANY_STRING,
+                applicable_to=TagApplicableTo.IMAGES_ONLY,
+            )
+            project_meta = project_meta.add_tag_meta(multiview_tag_meta)
+            need_update_project_meta = True
+        elif multiview_tag_meta.sly_id is None:
+            logger.warning(f"`sly_id` is None for group tag, trying to get it from server")
+            need_update_project_meta = True
+
+        if multiview_tag_meta.value_type != TagValueType.ANY_STRING:
+            raise ValueError(f"Tag '{multiview_tag_name}' is not of type ANY_STRING.")
+        elif multiview_tag_meta.applicable_to == TagApplicableTo.OBJECTS_ONLY:
+            raise ValueError(f"Tag '{multiview_tag_name}' is not applicable to images.")
+
+        if need_update_project_meta:
+            project_meta = self._api.project.update_meta(id=project_id, meta=project_meta)
+            context["project_meta"] = project_meta
+            multiview_tag_meta = project_meta.get_tag_meta(multiview_tag_name)
+
+        if not project_meta.project_settings.multiview_enabled:
+            if multiview_tag_name == _MULTIVIEW_TAG_NAME:
+                self._api.project.set_multiview_settings(project_id)
+            else:
+                self._api.project._set_custom_grouping_settings(
+                    id=project_id,
+                    group_images=True,
+                    tag_name=multiview_tag_name,
+                    sync=False,
+                )
+            project_meta = ProjectMeta.from_json(
+                self._api.project.get_meta(project_id, with_settings=True)
+            )
```

Comment:
Mb update `_set_custom_grouping_settings` to return meta, this will help to get the updated meta without creating a new request.

Reply:
`_set_custom_grouping_settings` / `set_multiview_settings` methods update only the project settings. They do not have access to actual meta, and extra request is still need.